sky/resources.py

```diff
@@ -275,7 +275,35 @@ def _set_region_zone(self, region: Optional[str],
         self._region, self._zone = self._cloud.validate_region_zone(
             region, zone)
 
-    def _try_validate_instance_type(self):
+    def get_valid_region_zones(self) -> List[Tuple[clouds.Region, clouds.Zone]]:
+        """Returns a list of (region, zone) that can provision this Resources"""
+        assert self.is_launchable()
+
+        if isinstance(self._cloud, clouds.GCP):
+            gcp_region_zones = list(
+                self._cloud.region_zones_provision_loop(
+                    instance_type=self._instance_type,
+                    accelerators=self.accelerators,
+                    use_spot=self._use_spot))
+
+            # GCP provision loop yields 1 zone per request.
+            # For consistency with other clouds, we group the zones in the
+            # same region.
```

Comment:
Is this required or an optimization for `_generate_launchables_with_region_zones()`? Should document.

Reply:
It is required. Added a comment.